File: College/13.c
/*

> Date Created: October 15, 2023
> Author: Ishaan Rastogi
> Purpose: WAP in C to calculate the root of a Quadratic Equation
> Operating System: This is only for Windows OS, it may or may not work on other OS
> Program Status: 100% Working

*/

#include <stdio.h>
#include <math.h>

int main(void)
{
    float a, b, c, discriminant, root1, root2;
    printf("Enter the coefficients of the quadratic equation (a, b, c): ");
    scanf("%f %f %f", &a, &b, &c);
    discriminant = b*b - 4*a*c;

    if (discriminant < 0)
    {
        printf("No real roots exist.\n");
    }
    else if (discriminant == 0)
    {
        root1 = -b / (2 * a);
        printf("The root is: %.2f\n", root1);
    }
    else
    {
        root1 = (-b + sqrt(discriminant)) / (2 * a);
        root2 = (-b - sqrt(discriminant)) / (2 * a);
        printf("The roots are: %.2f and %.2f\n", root1, root2);
    }
}

/* 

Ctrl + Shift + `

Terminal Code Loading...

gcc 13.c
./a.exe
 
*/

File: Web Development Projects/15 Intermediate JS/13.js
/*
> Date Created: June 10, 2024
> Author: Ishaan Rastogi
> Purpose: Fibonacci Numbers Generator
> Operating System: This is only for Windows OS, it may or may not work on other OS
> Program Status: 100% Working
*/

/*

0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144 ....

Series of last 2 addends, Eg - 5 + 8 = 13 

Generate with an output of n = 5.

*/

function fibonnaciGenerator(n) {
    var output = [];

    if ( n === 1) {
        output = [0];
    }
    else if ( n === 2 ) {
        output = [0,1];
    }
    else {
        output = [0,1];
        for ( var i = 2; i < n; i++ ){
            output.push(output[output.length-2] + output[output.length-1]); //pushing back to the output
        }
    }

    return output;
}

output = fibonnaciGenerator(5);
console.log(output);

/* 
For Terminal Code... Ctrl + Shift + `
*/